src/fabric/src/fabric2_db.erl

```diff
@@ -582,6 +591,44 @@ get_missing_revs(Db, JsonIdRevs) ->
     {ok, AllMissing}.
 
 
+validate_docid(<<"">>) ->
```

Comment:
This version of validate_docid is dropping behavior given its not using the same logic in `couch_doc:validate_doc_id/2` which has logic gated on the database name.

Reply:
```
validate_docid(DocId, DbName) ->
    case DbName =:= ?l2b(config:get("mem3", "shards_db", "_dbs")) andalso
        couch_db:is_system_db_name(DocId) of
        true ->
            ok;
        false ->
            validate_docid(DocId)
    end.
```

it’s for checking if we are writing to shards db and we don’t have a shards db

Reply:
Fair enough!